### ISSUE TITLE
`y_text.format` edge case

### ISSUE THREAD

**Issue by Waidhoferj** [y-crdt/y-crdt#101]
Testing with `ywasm` I found some interesting behavior with `format()`. Seems like the delta returned from the observer only picks up changes when the last character is included in the formatting block. For example:

```javascript
const x = d1.getText('test')
let target = null
let delta = null
let observer = x.observe(e => {
    target = e.target
    delta = e.delta
})

transact(d1,txn => x.insert(txn, 0, 'abcd'))

let start = 1
let len = 3
transact(d1,txn => x.format(txn, start, len, { bold: true }))
console.log(delta)
```

When `len = 3` I get the expected delta: `[ { retain: 1 }, { retain: 3, attributes: { bold: true } } `
But when `len = 2` the delta is an empty array: `[]`
I'd expect something like `[ { retain: 1 }, { retain: 2, attributes: { bold: true } } `

Is this correct behavior?

**Comment by Horusiath** [comment]
It looks like a bug in a format method - I've recreated this scenario in tests. Basically after formatting block store has following shape:

```
(<1#0>, len: 1, parent: null, right: <1#4>: 'a')
(<1#1>, len: 2, parent: null, origin-l: <1#0>, left: <1#5>, right: <1#3>: 'bc')
(<1#3>, len: 1, parent: null, origin-l: <1#2>, left: <1#1>: 'd')
(<1#4>, len: 1, parent: null, origin-l: <1#0>, origin-r: <1#1>, left: <1#0>, right: <1#5>: '<bold=true>')
(<1#5>, len: 1, parent: null, origin-l: <1#4>, origin-r: <1#3>, left: <1#4>, right: <1#1>: '<bold=null>')
```

Issue is more visible when calculating `event.toDelta` - this is an order in which blocks where traversed when computing delta:

```
<1#0> - visited string: 'a'
<1#4> - visited format: bold->true
<1#5> - visited format: bold->null
<1#1> - visited string: 'bc'
<1#3> - visited string: 'd'
```

**tl;dr** applying `ytext.format` caused is supposed to create text block: `a<bold>bc</bold>d`, but it created `a<bold></bold>bcd`. I'll make a PR to fix that.
